### ISSUE TITLE
Null Default DataFetchers will overwrite non null ones inside RuntimeWiring.Builder

### ISSUE THREAD

**Issue by AntoineGRHD** [graphql-java/graphql-java#3431]
**Describe the bug**

Successive calls to RuntimeWiring.Builder::type will overwrite the defaultDataFetchers value with null if the defaultDataFetcher isn't present.

```Java 
wiringBuilder.type("Query", builder -> builder.defaultDataFetcher(defaultDataFetcher)) //Sets the default dataFetcher
             .type("Query", builder -> builder.dataFetcher("asset", assetDataFetcher)) //Unsets the default dataFetcher
             .type("Query", builder -> builder.dataFetcher("person", personDataFetcher))
 ```

Setting the default data fetcher with the last call will work, but I believe this isn't a good solution since this is a builder pattern and there's already a null check for setting the enumValuesProvider, and typeResolver.

Moreover in my case using it with spring-graphql, Spring's RuntimeWiringConfigurer will first let you set your custom DataFetchers and then add all the annotated data fetchers, not the other way around.

**Proposed solution**

Inside `graphql.schema.idl.RuntimeWiring.class, RuntimeWiring.Builder::type(TypeRuntimeWiring typeRuntimeWiring)`

Replace :

```Java
this.defaultDataFetchers.put(typeName, typeRuntimeWiring.getDefaultDataFetcher());
```

With : (Just like the typeResolver and enumValuesProvider right after that)

```Java
DataFetcher defaultDataFetcher = typeRuntimeWiring.getDefaultDataFetcher();
if (defaultDataFetcher != null) {
        this.defaultDataFetchers.put(typeName, defaultDataFetcher);
}
```

**Comment by bbakerman** [comment]
Why is that you calling a `TypeRuntimeWiring.Builder` so many times?   Is this trying to show that multiple steps might continue to add more config into an existing type?

eg your example 

```java
wiringBuilder.type("Query", builder -> builder.defaultDataFetcher(defaultDataFetcher)) //Sets the default dataFetcher
             .type("Query", builder -> builder.dataFetcher("asset", assetDataFetcher)) //Unsets the default dataFetcher
             .type("Query", builder -> builder.dataFetcher("person", personDataFetcher))
```

versus

```java
wiringBuilder.type("Query", builder -> builder
    .defaultDataFetcher(defaultDataFetcher)
    .dataFetcher("asset", assetDataFetcher)
    .dataFetcher("person", personDataFetcher)
)
```

??

I can look into this

**Comment by AntoineGRHD** [comment]
Thanks for working on this issue.

Concerning your question, you're absolutely right.
The RuntimeWiring.Builder is used by multiple beans in spring-graphql (through the RuntimeWiringConfigurer), each bean call `type()`. I wanted to convey that idea in my example.
I wouldn't do repeated calls to `type()` in the same method

**Comment by dondonz** [comment]
The fix for this issue was included in the v22.1 release, thanks for reporting the issue
